addon/lib/connect.js

```diff
@@ -0,0 +1,46 @@
+import Ember from 'ember';
+import reduxStore from '../services/redux-store';
+
+const {
+  computed,
+  defineProperty,
+  inject: { service }
+} = Ember;
+
+export default function connect(mapStateToComputed, mapDispatchToActions, storeService) {
+  return function wrapWithConnect(WrappedComponent) {
+    return WrappedComponent.extend({
+      reduxStore: storeService || service('reduxStore'),
+
+      init() {
+        this._super(...arguments);
+        this.actions = this.actions || {};
+        const propMap = Ember.typeOf(mapStateToComputed) === 'function' ? 
+          mapStateToComputed(reduxStore) :
```

Comment:
maybe just move `const store = …` up to the top and use the same reference in both places.